content/setting-up-tugboat/select-repo-settings.md

```diff
@@ -125,6 +126,12 @@ credentials from within the repo settings.
 
 ![Authenticate with a Docker registry](../../_images/authenticate-with-a-docker-registry-add-credentials.png)
 
+### Preview IP Filters
```

Comment:
This needs to have more information, which I can get to you when I have a chance

* How to format the config values
* How the filtering works
  * Default behavior
  * Behavior when one or more filters are present

Reply:
> How to format the config values

The config can accept any of the following

* Any valid IPv4 subnet, using CIDR notation
* Any valid IPv6 subnet, using CIDR notation
* Any valid IPv4 address. We will automatically append a /32 CIDR subnet mask
* Any valid IPv6 address. We will automatically append a /128 CIDR subnet mask

Information about CIDR notation: https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing

> How the filtering works

When no filters are provided, no filtering is done. Preview URLs are publicly accessible to anyone that has the link (this is how Tugboat has always worked)

If one or more filters are provided, ONLY requests coming from an address that matches one of those filters are allowed access to the Preview URLs. Any other requests will be given a 404 - Preview Not Found.

An address "matches" if it is a member of any of the subnets in the list of filters.

> Bonus, not sure if you want to write this up or not

We also support DENY filters, but those can only be managed by the API or CLI. _deny_ filters are processed AFTER _allow_ filters.

If both _allow_ and _deny_ filters are present, a source IP must match an _allow_ filter, but must NOT match a _deny_ filter in order to access a Preview URL. 

If only _deny_ filters are present, a source IP must NOT match a _deny_ filter in order to access a Preview URL.

_deny_ filters are not very useful on their own, but provide a way for advanced users to allow access to a large subnet, and then specify exceptions to that allow list.